mirascope/base/ops_utils.py

```diff
@@ -0,0 +1,161 @@
+import inspect
+from contextlib import AbstractContextManager
+from functools import wraps
+from typing import Any, Callable
+
+from pydantic import BaseModel
+
+from .calls import BaseCall
+
+
+def get_class_vars(cls: BaseModel) -> dict[str, Any]:
+    """Get the class variables of a `BaseModel` removing any dangerous variables."""
+    class_vars = {}
+    for classvars in cls.__class_vars__:
+        if not classvars == "api_key":
+            class_vars[classvars] = getattr(cls.__class__, classvars)
+    return class_vars
+
+
+def get_wrapped_call(call: Callable, cls: BaseCall, **kwargs) -> Callable:
+    """Wrap a call to add the `llm_ops` parameter if it exists."""
+
+    if cls.configuration.llm_ops:
+        wrapped_call = call
+        for op in cls.configuration.llm_ops:
+            wrapped_call = op(
+                wrapped_call,
+                cls._provider,
+                **kwargs,
+            )
+        return wrapped_call
+    return call
+
+
+def mirascope_span(
+    fn: Callable,
+    handle_before_call: Callable,
```

Comment:
we should (ideally) provider a more fleshed out type for these callables so that the user knows what it expects and gets editor support

Reply:
we will of course also need to provide detailed documentation for writing such a function, and we can likely use Jake's saving example as the doc example